activejob/lib/active_job/queue_adapters.rb

```diff
@@ -48,5 +48,32 @@ module QueueAdapters
     autoload :SneakersAdapter
     autoload :SuckerPunchAdapter
     autoload :TestAdapter
+
+    ADAPTER = 'Adapter'.freeze
+
+    class << self
+      def instance(name, *adapter_args, &block)
+        hash = ([name, block] << adapter_args).hash
```

Comment:
I think you are misunderstanding what I am worried about. How you read the discussion in #15458?

The problem is that this is allowed (though admittedly highly unlikely):
```ruby
['AdapterName', Proc.new {}, :arg1, :arg2].hash # => 12345
['DifferentAdapterName', Proc.new {}, :different_arg].hash # => 12345
```